pkg/apis/extensions/validation/validation.go

```diff
@@ -279,7 +282,7 @@ func ValidateDeploymentStrategy(strategy *extensions.DeploymentStrategy, fldPath
 		if strategy.RollingUpdate == nil {
 			allErrs = append(allErrs, field.Required(fldPath.Child("rollingUpdate"), "this should be defaulted and never be nil"))
 		} else {
-			allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, fldPath.Child("rollingUpdate"))...)
+			allErrs = append(allErrs, ValidateRollingUpdateDeployment(strategy.RollingUpdate, replicas, fldPath.Child("rollingUpdate"))...)
```

Comment:
Yes, I am asking explicitly in the issue that we shouldn't add this in validation.